app/models/ability.rb

```diff
@@ -0,0 +1,37 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Handle the case where we don't have a current_user i.e. the user is a guest
+    user ||= User.new
+
+    return unless user.is? :admin
+
+    can :destroy, Post
+    can :destroy, Comment
```

Comment:
- [ ] As stated by the previous reviewer, you should add role for the user can manage his post and comments not other user can do this. You can make this by adding the roles like the following example:
```ruby
 if user.is? :admin
      can :manage, :all
    else
      can :manage, Post, author_id: user.id
      can :manage, Comment, author_id: user.id
      can :manage, Like, author_id: user.id
      can :read, :all
    end
```

Reply:
Hello @omarsalem7, 

I did it. return unless user.is> :admin is shorthand form of the if condition you stated. You can check it out.